sdks/python/apache_beam/runners/portability/flink_runner_test.py

```diff
@@ -427,11 +427,29 @@ def test_sql(self):
 
 
 class FlinkRunnerTestStreaming(FlinkRunnerTest):
+  def __init__(self, *args, **kwargs):
+    super(FlinkRunnerTestStreaming, self).__init__(*args, **kwargs)
+    self.enable_commit = False
+
+  def setUp(self):
+    self.enable_commit = False
+
   def create_options(self):
     options = super(FlinkRunnerTestStreaming, self).create_options()
     options.view_as(StandardOptions).streaming = True
+    if self.enable_commit:
+      options._all_options['checkpointing_interval'] = 3000
+      options._all_options['shutdown_sources_after_idle_ms'] = 60000
     return options
 
+  def test_callbacks_with_exception(self):
+    self.enable_commit = True
+    super(FlinkRunnerTest, self).test_callbacks_with_exception()
+
+  def test_register_finalizations(self):
+    self.enable_commit = True
+    super(FlinkRunnerTest, self).test_register_finalizations()
```

Comment:
I've been thinking that we should support bundle finalization also with checkpointing disabled in an eager fashion. Does not have to be tackled here but would be nice to fix eventually because checkpointing is disabled by default.

Reply:
If we can have bundle finalization not rely on checkpoint, that would be really good. I'm curious does it only require changes in Beam or it also requires some updates on Flink side like the batch support we need mentioned in https://lists.apache.org/thread.html/r89c74f4020cd15d3c1a4b84ef78d7da4f67824e1b5e2210f4c5c1f40%40%3Cdev.flink.apache.org%3E.

Reply:
It definitely requires changes in Flink for the finalization to be persisted without checkpointing by restoring the state from materialized intermediate results. However, if the user is fine with the job re-executing in case of failures, we may still offer bundle finalization. That may be acceptable as opposed to piling up to-be-acknowledged bundles which is likely going to fail the job. Perhaps this should be opt-in.